docs/interfaces.md

```diff
@@ -0,0 +1,571 @@
+        [out, size=max_len] uint8_t *values, uint32_t max_len, [out] uint32_t *values_len,
+        [in, out] sgx_cmac_128bit_tag_t *cmac,
+        [user_check] void *u_shim_ctx);
+
+```
+
+```c++
+// verify msp with the help of tlcc
+void ocall_tlcc_request();
+```
+
+```c++
+// Provided/implemented by a common sealed storage module
+void ocall_load(
+        [in, string] const char *str, 
+        [out, size=max_data_len] uint8_t *data, uint32_t max_data_len
+        [out] uint32_t *data_len);
+
+void ocall_store(
+        [in, string] const char *str, 
+        [in, size=data_len] uint8_t *data, uint32_t data_len);
+```
+
+
+Additionally, ECC includes a common attestation module that abstracts
+
+```c++
+// attestation
+void ocall_get_spid(); // TBD: is spid only needed for EPID?
+
+// creates an attestation evidence; the evidence_request includes all the form of evidence (EPID/DCAP), the quote, and additional information required to complete the evidence request
+void ocall_create_evidence(
+        [in] evidence_request
+        [out] evidence
+);
+```
+
+## State:
+An ECC instance keeps a local reference of the FPC chaincode enclave id.
+
+```go
+type EnclaveChaincode struct {
+        erccStub ercc.EnclaveRegistryStub
+        tlccStub tlcc.TLCCStub
+        enclave  enclave.Stub
+        verifier crypto.Verifier
+        ev       attestation.EvidenceService // abstracts evidence creation for EPID/DCAP
+}
+
+type StubImpl struct {
+	eid C.enclave_id_t
+}
+```
+
+
+# ECC_Enclave
+
+## Interface:
+The ECC_Enclave interface specifies interface of an FPC chaincode enclave. 
+
+Note: `sealed credentials` is renamed to `sealed state`
+
+```c++
+// initializes an enclave
+public int ecall_init(
+        [in, size=cc_params_len] uint8_t *cc_params, uint32_t cc_params_len,
+        [out] uint8_t *credentials);     // TBD is this a fixed length type?
+
+// invoke a FPC chaincode
+public int ecall_chaincode_invoke(
+        [in, size=proposal_len]  uint8_t *proposal, uint32_t proposal_len,
+        [out, size=response_len_in] uint8_t *response, uint32_t response_len_in,
+        [out] uint32_t *response_len_out,
+        [out] sgx_ec256_signature_t *signature,
+        [user_check] void *u_shim_ctx);
+
+// key generation and distribution
+// TODO interface below needs more brain cycles
+public int ecall_generate_cc_keys(
+        // TBD: Why do we need cc_params?
+        [in, size=cc_params_len] uint8_t *cc_params, uint32_t cc_params_len, 
+        [out] uint8_t *cckey_registration_msg,
+        [out] uint8_t *sealed_cckeys);
+
+public int ecall_export_cc_keys(
+       [in] uint8_t *target_enclave_vk,
+       [out] uint8_t *export_msg);
+
+public int ecall_import_cc_keys(
+        // TBD: Why do we need cc_params?
+        [in, size=cc_params_len] uint8_t *cc_params, uint32_t cc_params_len, 
+        [in] uint8_t *export_msg,
+        [out] uint8_t *cckey_registration_msg,
+        [out] uint8_t *sealed_cckeys);
```

Comment:
if we have a sealed storage module, then there would also not be any return here? (similar to `init`)?

Reply:
we still want to return `export_msg` as this is part of the protocol and later stored at ercc. Removed all sealed messages as they are stored using the storage module.

Reply:
right, edited

Reply:
there are still some old references to ecall_create_enclave_with_sealed_credentials and other references to sealed state sprinkled around.  If you want to keep some (potential) ocalls/ecalls and reference to edl here, rather than just move to #421, then maybe having a separate section where all of it is consolidated would be clearer (and better manageable/kept consistent)?